src/Multipart/AbstractUploadManager.php

```diff
@@ -49,16 +48,21 @@
     /** @var UploadState State used to manage the upload. */
     protected $state;
 
+    /** @var string Exception type thrown for the upload */
+    protected $exception;
+
     /**
      * @param Client $client
+     * @param string $exception
      * @param array  $config
      */
-    public function __construct(Client $client, array $config = [])
+    public function __construct(Client $client, array $config = [], $exception = null)
```

Comment:
Why not use `$config` instead? It's a bit awkward to put a scalar argument after an associative array argument. Can you also call it `exceptionClass` if an argument or `exception_class` if a config value?